src/Table/TableController.cc

```diff
@@ -202,7 +202,24 @@ void TableController::OnFileListRequest(
     file_list_response.set_parent(parent_path.string());
 
     try {
+        // get total number of files in the directory
+        using std::filesystem::directory_iterator;
+        using func_ptr = bool (*)(const std::filesystem::path&);
+        auto total_files = std::count_if(directory_iterator(file_path), directory_iterator{}, (func_ptr)std::filesystem::is_regular_file);
```

Comment:
not sure why you need to use `using` above. It doesn't really help anything. Why not just provide the types in the `count_if`?

Reply:
i.e. `auto total_files = std::count_if(fs::directory_iterator(file_path), directory_iterator{}, (bool (*)(const fs::path&))fs::is_regular_file);;`

Reply:
CASA images are folders, so I do wonder whether we should actually include _all_ directory entries, not just regular files